ArraySimulation/MPPT/GlobalMPPTAlgorithms/VoltageSweep.py

```diff
@@ -0,0 +1,137 @@
+"""
+VoltageSweep.py
+
+Author: Afnan Mir, Array Lead (2021).
+Contact: afnanmir@utexas.edu
+Created: 02/06/2021
+Last Modified: 02/08/2021
+
+Description: The Voltage Sweep class is a derived concrete class of
+GlobalAlgorithm implementing the Voltage Sweep algorithm. It increments through
+the range of all possible voltage values (the "sweep"), finding all local maxima
+of the P-V curve. It then identifies the global maxima using a LocalMPPTAlgorithm.
+"""
+# Library Imports.
+
+
+# Custom Imports.
+from ArraySimulation.MPPT.GlobalMPPTAlgorithms.GlobalMPPTAlgorithm import (
+    GlobalMPPTAlgorithm,
+)
+
+
+class VoltageSweep(GlobalMPPTAlgorithm):
+    """
+    The Voltage Sweep class is a derived concrete class of GlobalAlgorithm
+    implementing the Voltage Sweep algorithm. It increments through the range of
+    all possible voltage values (the "sweep"), finding all local maxima of the
+    P-V curve. It then identifies the global maxima using a LocalMPPTAlgorithm.
+    """
+
+    def __init__(self, numCells=1, MPPTLocalAlgoType="Default", strideType="Fixed"):
+        super(VoltageSweep, self).__init__(
+            numCells, "Voltage Sweep", MPPTLocalAlgoType, strideType
+        )
+
+        # Stores all the voltage values of the local maxima.
+        self.voltage_peaks = [0]
+
+        # Stores the power values of the local maxima.
+        self.power_peaks = [0]
+
+        # Whether we are in sweeping mode or not.
+        self.sweeping = True
+
+        # Checks to see if we were increasing before.
+        self.increasing = True
+
+        self.stride = 0.01
```

Comment:
These can be removed since they were declared in the parent class. Stride can be an instance variable of the parent class as well. We've left a TODO in line 68 of the GlobalMPPTAlgorithm class anyway to add it as a parameter.

Reply:
These instance variables are not in the parent GlobalMPPTAlgorithms class. It is specific to the VoltageSweep Class